lib/definition.js

```diff
@@ -197,38 +155,50 @@ module.exports = class Definition {
     , 'max': this._max
     , 'list_type': this._list_type
     , 'separator': this._list_separator
+    , 'allow_empty': this._allow_empty
     }
   }
 
   _populate() {
-    if (!has(process.env, this._env)) {
-      if (this._default !== null) {
-        this._value = this._transform(this._default)
-      }
+    const value_is_missing = !has(process.env, this._env)
+      || process.env[this._env] === EMPTY_STRING
 
-      if (this._required) {
-        this._error = new MissingEnvError({
+    if (this._required) {
+      if (value_is_missing) {
+        const err = new MissingEnvError({
           type: this._type
         , name: this._env
         })
+        throw err
+      }
+    }
+
+    if (value_is_missing) {
+      if (this._default !== null && !this._allow_empty) {
```

Comment:
is it worth mentioning somewhere that `allowEmpty` only has effect when there is a `default()`? I could see that being confusing - empty values are always "allowed", this just controls if you can override a default with empty.

Reply:
Probably should mention that, yeah.  After all, you don't need `allowEmpty` if there's no default.